master/static/migrations/20240304145201_add-models-trigger.tx.up.sql

```diff
@@ -0,0 +1,71 @@
+-- sequence for tracking event order
+CREATE SEQUENCE IF NOT EXISTS stream_model_seq START 1;
+
+ALTER TABLE models ADD COLUMN IF NOT EXISTS seq bigint DEFAULT 0;
+
+-- trigger function to update sequence number on row modification
+-- this should fire BEFORE so that it can modify NEW directly.
+CREATE OR REPLACE FUNCTION stream_model_seq_modify() RETURNS TRIGGER AS $$
+BEGIN
+    NEW.seq = nextval('stream_model_seq');
+RETURN NEW;
+END;
+$$ LANGUAGE plpgsql;
+
+DROP TRIGGER IF EXISTS stream_model_trigger_seq ON models;
+CREATE TRIGGER stream_model_trigger_seq
+    BEFORE INSERT OR UPDATE OF
+    name, description, creation_time, last_updated_time, metadata, labels, user_id, archived, notes, workspace_id
```

Comment:
it's likely outside the scope of this ticket; however, due to the limitations of postgres notify/listen queues, we'll want to only communicate over these channels using filterable columns. It doesn't look like the work on making the transition to the strategy is complete.

That being said, this will need to be adjusted with the work so that we don't potentially exceed that 8k character limit with user defined fields

Reply:
For models, I think it should be safe to exclude `creation_time` and `user_id` from the trigger, but do you mean we also want to exclude `metadata` and `notes`? I know we have a [ticket](https://hpe-aiatscale.atlassian.net/browse/MD-266) to address the character limit, I wonder if that's still valid, maybe @jgongd has more information?

Reply:
The ticket is related to this issue. I plan to add unit tests next before focusing on this ticket. Let me know if you think this should take priority, and I'll be happy to tackle it first.